sql/parser/types.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package parser
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+const (
+	astInt        = "INT"
+	astInteger    = "INTEGER"
+	astTinyInt    = "TINYINT"
+	astSmallInt   = "SMALLINT"
+	astMediumInt  = "MEDIUMINT"
+	astBigInt     = "BIGINT"
+	astReal       = "REAL"
+	astDouble     = "DOUBLE"
+	astFloat      = "FLOAT"
+	astDecimal    = "DECIMAL"
+	astNumeric    = "NUMERIC"
+	astChar       = "CHAR"
+	astVarChar    = "VARCHAR"
+	astBinary     = "BINARY"
+	astVarBinary  = "VARBINARY"
+	astText       = "TEXT"
+	astTinyText   = "TINYTEXT"
+	astMediumText = "MEDIUMTEXT"
+	astLongText   = "LONGTEXT"
+	astBlob       = "BLOB"
+	astTinyBlob   = "TINYBLOB"
+	astMediumBlob = "MEDIUMBLOB"
+	astLongBlob   = "LONGBLOB"
+)
+
+// ColumnType represents a type in a column definition.
+type ColumnType interface {
+	fmt.Stringer
+	columnType()
+}
+
+// BitType represents a BIT type.
+type BitType struct {
+	N int
+}
+
+func (*BitType) columnType() {}
+
+func (node *BitType) String() string {
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, "BIT")
+	if node.N >= 0 {
+		fmt.Fprintf(&buf, "(%d)", node.N)
+	}
+	return buf.String()
+}
+
+// IntType represents an INT, TINYINT, SMALLINT, MEDIUMINT, BIGINT or INTEGER
+// type.
+type IntType struct {
+	Name     string
+	N        int
```

Comment:
`N` is the formatting width, not the value being stored. `IntType` is a descriptor for the type, not a holder of a particular value.